js/base-util.js

```diff
@@ -105,52 +100,179 @@ function baseUtil () {
+				top: 0px;
+				height: 100px;
+				transition: height 2s;
+				overflow: hidden;
+				border-bottom: 1px solid rgb(32, 194, 14);
+			}
+			.lamer-chat > span {
+				display: block; white-space: nowrap;
+				padding: 0px 5px 0px 45px;
+			}
+			.lamer-chat > span > span {
+				float: left; margin-left: -39px;
+			}
+			</style>
+		</div>`;
+		$(`#textchat`).prepend(consTemplate);
+		let blink = false;
+		d20plus.ut.cursor = setInterval(() => {
+			$(`.lamer-chat`).append($(`.lamer-cursor`));
+			if (blink) $(`.lamer-chat aside`).html("|");
+			else $(`.lamer-chat aside`).html("");
+			blink = !blink;
+		}, 300);
 	};
 
+	d20plus.ut.showLoadingMessage = () => {
+		const isStreamer = !!d20plus.cfg?.get("chat", "streamerChatTag");
+		const scriptName = isStreamer ? "Script" : d20plus.scriptName;
+		const loadmsgtemplate = `<span><span>&gt;</span>loading ${d20plus.scriptName}</span>`;
+		if (!isStreamer) $(".lamer-chat > span:first-child").after(loadmsgtemplate);
+		if (!window.enhancementSuiteEnabled) d20plus.ut.showHardDickMessage(scriptName);
+	}
+
 	d20plus.ut.sendHackerChat = (message, error = false) => {
+		const legacytStyle = !!d20plus.cfg.get("chat", "legacySystemMessagesStyle");
+		if (!message) return;
 		d20.textchat.incoming(false, ({
 			who: "system",
-			type: "system",
-			content: `<span class="${error ? "hacker-chat-error" : "hacker-chat"}">
+			type: !legacytStyle && error ? "error" : "system",
+			content: (legacytStyle ? `<span class="${error ? "hacker-chat-error" : "hacker-chat"}">
 				${message}
-			</span>`,
+			</span>` : message),
 		}));
 	};
 
+	d20plus.ut.plantVersionInfo = () => {
+		const thisPlayer = d20?.Campaign.players.get(d20_player_id);
+		if (!thisPlayer) return;
+		if (!d20plus.cfg.getOrDefault("chat", "shareVersions")) {
+			if (thisPlayer.get("script")) {
+				thisPlayer.set("script", null, true);
+				thisPlayer.save();
+			}
+			return;
+		}
+		d20plus.ut.log("Managing version info");
+		const b20n = encodeURI(d20plus.scriptName.split("-")[1].split(" v")[0]);
+		const b20v = encodeURI(d20plus.version);
+		const vtte = encodeURI(window.r20es?.hooks?.welcomeScreen?.config?.previousVersion);
+		const phdm = d20plus.ut.detectDarkModeScript();
+		const date = Number(new Date());
+		const info = btoa(JSON.stringify({b20n, b20v, vtte, phdm, date}));
+		thisPlayer.set("script", info, true);
+		thisPlayer.save();
+	}
+
+	d20plus.ut.cmpVersions = (present, latest) => {
+		if (!present || !latest) return 0;
+		const regExStrip0 = /(\.0+)+$/;
+		const segmentsA = present.replace(regExStrip0, "").split(".");
+		const segmentsB = latest.replace(regExStrip0, "").split(".");
+		const l = Math.min(segmentsA.length, segmentsB.length);
+
+		for (let i = 0; i < l; i++) {
+			const diff = parseInt(segmentsA[i], 10) - parseInt(segmentsB[i], 10);
+			if (diff) {
+				return diff;
+			}
+		}
+		return segmentsA.length - segmentsB.length;
+	}
+
+	d20plus.ut.detectDarkModeScript = () => {
+		d20plus.ut.dmscriptDetected = false;
```

Comment:
stick a comment with a link to said script in here

Reply:
Done